src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -2003,6 +2003,11 @@ public static AccessMode valueFromMethodName(String methodName) {
                 default -> throw new IllegalArgumentException("No AccessMode value for method name " + methodName);
             };
         }
+
+        private static final @Stable AccessMode[] VALUES = values();
+        static AccessMode valueFromOrdinal(int mode) {
+            return VALUES[mode];
+        }
```

Comment:
Also, I'll throw this out there, though I'm not sure how big of an issue it is: this array creation might have an effect on startup. But, since it is only used on a slow path right before we throw an exception (at least, at the moment), calling `values()` every time inside the `valueFromOrdinal` method, and avoiding the array creation on startup, might be 'better'.

Reply:
I think this is a worthy change: see https://github.com/openjdk/jdk/pull/14943/files#diff-556b309ea2df3f5bfe8229de944183ef19750ce4511d0328cd59af2ce2b61ae2R135
that `VarHandle.AccessMode.values()` is called every time a polymorphic VH call is linked.

If the startup cost from array allocation is really problematic, we can make it stable and allocate `VALUES` on demand (at `valueFromOrdinal` calls).

Reply:
Ah, so it looks like we are creating this array already, in adding the `valueFromOrdinal` method allows sharing. That sounds good!

Reply:
The overhead is just one time allocation of an array of length 31, which is a very small array which I don't think such small array creation impacts as much.

I should have replaced other uses of `AccessMode.values()`.   I will include that change in this patch. 

@liach  I don't know about `AccessType.values()` which seems to me not worth the change to add `valueFromOrdinal` as only calls that 6 times.